File: CompanionApps/iOS/Sources/Handlers/TextHandler.swift
/// Handles text-related gRPC requests by delegating to XCUITestBridge.
final class TextHandler: @unchecked Sendable {
    private let bridge: XCUITestBridge

    init(bridge: XCUITestBridge) {
        self.bridge = bridge
    }

    func typeText(_ text: String) async {
        await bridge.typeText(text)
    }

    func clearText(characterCount: Int?) async {
        await bridge.clearText(characterCount: characterCount)
    }

    func setText(
        id: String?,
        label: String?,
        containsText: String?,
        text: String,
        bundleID: String?,
        candidateBundleIDs: [String]
    ) async -> Bool {
        await bridge.setText(
            id: id,
            label: label,
            containsText: containsText,
            text: text,
            bundleID: bundleID,
            candidateBundleIDs: candidateBundleIDs
        )
    }
}

File: CompanionApps/Android/app/src/androidTest/java/com/amoo/companion/handlers/TextHandler.kt
package com.amoo.companion.handlers

import com.amoo.companion.bridge.UIAutomatorBridge

class TextHandler(private val bridge: UIAutomatorBridge) {

    fun typeText(text: String) {
        bridge.typeText(text)
    }

    fun clearText() {
        bridge.clearText()
    }

    fun setText(resourceId: String?, label: String?, containsText: String?, value: String): Boolean {
        return bridge.setText(resourceId, label, containsText, value)
    }
}
